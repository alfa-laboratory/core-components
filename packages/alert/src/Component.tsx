import React, { FC } from 'react';
import cn from 'classnames';

import { Plate, PlateProps } from '@alfalab/core-components-plate';

import styles from './index.module.css';

export type AlertProps = Omit<PlateProps, 'foldable' | 'defaultFolded' | 'leftAddons'>;

export const Alert: FC<AlertProps> = ({ className, title, ...restProps }) => (
    <Plate
        className={cn(styles.component, className)}
        buttonsClassName={styles.buttons}
        title={title ? <span className={styles.title}>{title}</span> : null}
        {...restProps}
    />
);
