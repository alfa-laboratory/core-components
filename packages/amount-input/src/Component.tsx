import cn from 'classnames';
import React, { useState, useEffect, useCallback, forwardRef } from 'react';
import { Input, InputProps } from '@alfalab/core-components-input';
import { FormControl } from '@alfalab/core-components-form-control';
import { THINSP, formatAmount, getCurrencySymbol } from '@alfalab/utils';
import { CurrencyCodes } from '@alfalab/data';
import { getFormatedValue, getAmountValueFromStr } from './utils';

import styles from './index.module.css';

export type AmountInputProps = Omit<InputProps, 'value' | 'onChange' | 'type'> & {
    /**
     * Денежное значение в минорных единицах
     * Значение null - значит не установлено
     */
    value?: number | null;

    /**
     * Валюта
     */
    currency?: CurrencyCodes;

    /**
     * Минорные единицы
     */
    minority?: number;

    /**
     * Жир
     */
    bold?: boolean;

    /**
     * Обработчик события изменения значения
     */
    onChange?: (
        e: React.ChangeEvent<HTMLInputElement>,
        payload: {
            /**
             * Денежное значение в минорных единицах
             * Значение null - значит не установлено
             */
            value: number | null;
            /**
             * Значение инпута
             */
            valueString: string;
        },
    ) => void;
};

/**
 * Компонент для ввода денежных значений
 */
export const AmountInput = forwardRef<HTMLInputElement, AmountInputProps>(
    (
        {
            value = null,
            minority = 100,
            currency = 'RUR',
            placeholder = `0\u2009${getCurrencySymbol(currency) || ''}`,
            bold = true,
            className,
            focusedClassName,
            dataTestId,
            clear = false,
            onChange,
            onBlur,
            onFocus,
            onClear,
            ...restProps
        },
        ref,
    ) => {
        const [focused, setFocused] = useState(false);
        const [inputValue, setInputValue] = useState<string>(
            formatAmount({
                // TODO: поддержать nullable в utils
                value: value as number,
                currency,
                minority,
            }).formated,
        );
        const filled = Boolean(inputValue || focused);

        const currencySymbol = getCurrencySymbol(currency);

        useEffect(() => {
            const currentAmountValue = getAmountValueFromStr(inputValue, minority);
            if (currentAmountValue !== value) {
                return setInputValue(
                    formatAmount({
                        // TODO: поддержать nullable в utils
                        value: value as number,
                        currency,
                        minority,
                    }).formated,
                );
            }

            return () => undefined;
            // eslint-disable-next-line react-hooks/exhaustive-deps
        }, [value, currency, minority]);

        const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
            const input = e.target;
            const enteredValue = input.value.replace(/\s/g, '').replace('.', ',');
            const isCorrectEnteredValue = /(^[0-9]{1,9}(,([0-9]+)?)?$|^\s*$)/.test(enteredValue);

            if (isCorrectEnteredValue) {
                const newFormatedValue = getFormatedValue(enteredValue, currency, minority);

                if (newFormatedValue === inputValue) {
                    const caret = input.selectionStart;
                    window.requestAnimationFrame(() => {
                        input.selectionStart = caret;
                        input.selectionEnd = caret;
                    });
                } else {
                    /**
                     * Поддержка положения каретки
                     * Поскольку при форматировании введенного значения могут появляться символы типа пробела
                     * или запятая - каретка прыгает в конец и ее необходимо ставить в правильное место
                     */

                    // Узнаем длину оригинального инпута с условием обрезания лишних символов

                    const [head, tail] = input.value.split(/\.|,/);
                    let notFormattedEnteredValueLength = head.length;
                    if (tail) {
                        notFormattedEnteredValueLength += 1; // запятая или точка
                        notFormattedEnteredValueLength += tail.slice(0, 2).length; // только 2 символа в минорной части
                    }

                    const diff = newFormatedValue.length - notFormattedEnteredValueLength;
                    const caret = (input.selectionStart as number) + diff;
                    window.requestAnimationFrame(() => {
                        input.selectionStart = caret;
                        input.selectionEnd = caret;
                    });
                }

                setInputValue(newFormatedValue);
                if (onChange) {
                    onChange(e, {
                        value: getAmountValueFromStr(newFormatedValue, minority),
                        valueString: newFormatedValue,
                    });
                }
            } else {
                // Не двигаем каретку когда вставляется невалидный символ
                const caret = (input.selectionStart as number) - 1;
                window.requestAnimationFrame(() => {
                    input.selectionStart = caret;
                    input.selectionEnd = caret;
                });
            }
        };

        const handleInputFocus = useCallback(
            (event: React.FocusEvent<HTMLInputElement>) => {
                setFocused(true);

                if (onFocus) {
                    onFocus(event);
                }
            },
            [onFocus],
        );

        const handleInputBlur = useCallback(
            (event: React.FocusEvent<HTMLInputElement>) => {
                setFocused(false);

                if (onBlur) {
                    onBlur(event);
                }
            },
            [onBlur],
        );

        const handleClear = useCallback(
            (event: React.MouseEvent<HTMLButtonElement>) => {
                setInputValue('');

                if (onClear) {
                    onClear(event);
                }
            },
            [onClear],
        );

        const [majorPart, minorPart] = inputValue.split(',');

        return (
            <div
                className={cn(styles.container, {
                    [styles.bold]: bold,
                    [styles.focused]: focused,
                    [styles.filled]: filled,
                })}
            >
                <FormControl
                    {...restProps}
                    className={cn(styles.fakeValueWithCurrencyContainer)}
                    focused={focused}
                    filled={filled}
                >
                    <div>
                        <span className={styles.majorPart}>{majorPart}</span>

                        <span className={styles.minorPartAndCurrency}>
                            {minorPart !== undefined && `,${minorPart}`}
                            {THINSP}
                            {currencySymbol}
                        </span>
                    </div>
                </FormControl>

                <Input
                    {...restProps}
                    clear={clear}
                    placeholder={placeholder}
                    value={inputValue}
                    className={cn(styles.component, className)}
                    focusedClassName={focusedClassName}
                    inputClassName={styles.input}
                    onChange={handleChange}
                    onFocus={handleInputFocus}
                    onBlur={handleInputBlur}
                    onClear={handleClear}
                    dataTestId={dataTestId}
                    ref={ref}
                />
            </div>
        );
    },
);
