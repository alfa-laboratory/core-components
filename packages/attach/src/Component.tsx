import React, {
    InputHTMLAttributes,
    ButtonHTMLAttributes,
    useState,
    useCallback,
    useRef,
} from 'react';
import cn from 'classnames';
import mergeRefs from 'react-merge-refs';
import { Button, ButtonProps } from '@alfalab/core-components-button';
import { ProgressBar } from '@alfalab/core-components-progress-bar';
import { KeyboardFocusable } from '@alfalab/core-components-keyboard-focusable';
import { AttachmentSBlackIcon, AttachmentMBlackIcon } from '@alfalab/icons-classic';
import { pluralize } from '@alfalab/utils';
import { truncateFilename } from './utils';

import styles from './index.module.css';

export type AttachProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'size' | 'type' | 'value'> & {
    /**
     * Содержимое кнопки для выбора файла
     */
    buttonContent?: React.ReactNode;

    /**
     * Свойства для кнопки
     */
    buttonProps?: ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Число символов, после которого имя файла будет обрезаться
     */
    maxFilenameLength?: number;

    /**
     * Текст для случая, когда файл не загружен
     */
    noFileText?: string;

    /**
     * Процент выполнения загрузки файла
     */
    progressBarPercent?: number;

    /**
     * Размер компонента
     */
    size?: 'xs' | 's' | 'm' | 'l';

    /**
     * Содержимое поля ввода, указанное по умолчанию. Принимает массив объектов типа File или null.
     */
    value?: File[] | null;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
};

const MULTIPLE_TEXTS: [string, string, string] = ['файл', 'файла', 'файлов'];

export const Attach = React.forwardRef<HTMLInputElement, AttachProps>(
    (
        {
            size = 's',
            accept,
            buttonContent = 'Выберите файл',
            buttonProps = {},
            className,
            disabled,
            dataTestId,
            id,
            maxFilenameLength,
            multiple,
            noFileText = 'Нет файла',
            progressBarPercent,
            value,
            onChange,
            ...restProps
        },
        ref,
    ) => {
        const [files, setFiles] = useState(value || []);

        const inputRef = useRef<HTMLInputElement>(null);
        const labelRef = useRef<HTMLLabelElement>(null);
        const buttonRef = useRef<HTMLButtonElement>(null);

        const handleInputChange = useCallback(
            (event: React.ChangeEvent<HTMLInputElement>) => {
                if (event.target.files) {
                    setFiles(Array.from(event.target.files));
                }

                if (onChange) {
                    onChange(event);
                }
            },
            [onChange],
        );

        const handleButtonClick = useCallback(
            event => {
                if (labelRef.current) {
                    labelRef.current.click();
                }
                if (buttonRef.current) {
                    buttonRef.current.focus();
                }

                if (buttonProps.onClick) {
                    buttonProps.onClick(event);
                }
            },
            [buttonProps],
        );

        const handleClearClick = useCallback(() => {
            if (inputRef.current) {
                inputRef.current.value = '';
            }

            setFiles([]);
        }, []);

        const Icon = size === 'xs' ? AttachmentSBlackIcon : AttachmentMBlackIcon;

        const statusTextContent =
            files.length === 1 ? (
                truncateFilename(files[0].name, maxFilenameLength)
            ) : (
                <abbr title={files.map(file => file.name).join()}>
                    {files.length} {pluralize(files.length, ...MULTIPLE_TEXTS)}
                </abbr>
            );

        return (
            <div
                className={cn(
                    cn(
                        styles.component,
                        styles[size],
                        {
                            [styles.disabled]: disabled,
                        },
                        className,
                    ),
                )}
            >
                <Button
                    {...buttonProps}
                    size={size}
                    disabled={disabled}
                    view={(buttonProps && buttonProps.view) || 'outlined'}
                    leftAddons={
                        (buttonProps && buttonProps.leftAddons) || (
                            <Icon className={cn(styles.icon)} />
                        )
                    }
                    onClick={handleButtonClick}
                    ref={buttonRef}
                >
                    <span>{buttonContent}</span>
                </Button>
                <label className={cn(styles.label)} htmlFor={id} ref={labelRef}>
                    <input
                        {...restProps}
                        className={cn(styles.control)}
                        accept={accept}
                        disabled={disabled}
                        id={id}
                        multiple={multiple}
                        tabIndex={-1}
                        type='file'
                        onChange={handleInputChange}
                        ref={mergeRefs([ref, inputRef])}
                        data-test-id={dataTestId}
                    />
                </label>
                {files && files.length > 0 ? (
                    <div className={cn(styles.file)}>
                        <span>{statusTextContent}</span>
                        <KeyboardFocusable>
                            {(ref, focused) => (
                                <button
                                    aria-label='очистить'
                                    type='button'
                                    className={cn(styles.clear, {
                                        [styles.focused]: focused,
                                    })}
                                    onClick={handleClearClick}
                                    ref={ref}
                                />
                            )}
                        </KeyboardFocusable>
                        {progressBarPercent && (
                            <ProgressBar
                                className={cn(styles.progressBar)}
                                value={progressBarPercent}
                                view='positive'
                            />
                        )}
                    </div>
                ) : (
                    <div className={cn(styles.noFile)}>{noFileText}</div>
                )}
            </div>
        );
    },
);

/**
 * Для отображения в сторибуке
 */
Attach.defaultProps = {
    size: 's',
    buttonContent: 'Выберите файл',
    noFileText: 'Нет файла',
};
