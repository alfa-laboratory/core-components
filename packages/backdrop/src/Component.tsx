import React, { MouseEvent } from 'react';
import cn from 'classnames';
import { TransitionProps } from 'react-transition-group/Transition';
import { CSSTransition } from 'react-transition-group';

import { CSSTransitionClassNames } from 'react-transition-group/CSSTransition';
import styles from './index.module.css';

export type BackdropProps = Partial<TransitionProps> & {
    /**
     * Прозрачный бэкдроп
     */
    invisible?: boolean;

    /**
     * Управляет видимостью компонента
     */
    open: boolean;

    /**
     * Обработчик клика по бэкдропу
     */
    onClose?: (event: MouseEvent<HTMLElement>) => void;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Классы анимации
     *
     * http://reactcommunity.org/react-transition-group/css-transition#CSSTransition-prop-classNames
     */
    transitionClassNames?: string | CSSTransitionClassNames;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;

    /**
     * Прозрачность бэкдропа
     */
    opacity?: number;
};

export const Backdrop: React.FC<BackdropProps> = ({
    className,
    open = false,
    invisible = false,
    timeout = 200,
    children,
    onClose,
    dataTestId,
    transitionClassNames = styles,
    opacity = 1,
    ...restProps
}) => {
    return (
        <CSSTransition
            timeout={timeout}
            unmountOnExit={true}
            classNames={transitionClassNames}
            in={open}
            appear={true}
            {...restProps}
        >
            <div
                aria-hidden={true}
                onClick={onClose}
                data-test-id={dataTestId}
                className={cn(styles.backdrop, className, {
                    [styles.invisible]: invisible,
                })}
                style={{ opacity }}
            >
                {children}
            </div>
        </CSSTransition>
    );
};
