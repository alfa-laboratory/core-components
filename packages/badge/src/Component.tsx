import React from 'react';
import cn from 'classnames';

import styles from './index.module.css';

export type BadgeProps = {
    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     *  Вид компонента
     */
    view: 'icon' | 'count';

    /**
     * Размер компонента (только для view=count)
     */
    size?: 's' | 'm' | 'l';

    /**
     *  Видимость обводки вокруг иконки (только для view=icon)
     */
    visibleIconOutline?: boolean;

    /**
     * Контент компонента
     */
    content?: React.ReactElement | number;

    /**
     * Цветовое оформление иконки
     */
    iconColor?:
        | 'positive'
        | 'attention'
        | 'link'
        | 'negative'
        | 'tertiary'
        | 'secondary'
        | 'primary';

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
};

export const Badge = ({
    className,
    size = 'm',
    view,
    visibleIconOutline = false,
    content,
    iconColor,
    dataTestId,
}: BadgeProps) => {
    const isCountView = view === 'count';

    const isHidden = isCountView && typeof content === 'number' && content <= 0;
    const componentContent = isCountView && content && content >= 100 ? '99+' : content;

    return (
        <div
            className={cn(
                styles.component,
                styles[size],
                styles[view],
                iconColor && styles[iconColor],
                {
                    [styles.isHidden]: isHidden,
                    [styles.dot]: !content,
                    [styles.outline]: visibleIconOutline,
                },
                className,
            )}
            data-test-id={dataTestId}
        >
            {componentContent}
        </div>
    );
};
