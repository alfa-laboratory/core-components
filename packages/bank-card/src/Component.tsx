import React, { useCallback, MouseEvent, ReactNode, useState, ChangeEvent, useEffect } from 'react';
import cn from 'classnames';
import { MaskedInput } from '@alfalab/core-components-masked-input';
// Дождаться иконку альфы в icons-logotype
import { BankAlfaLColorIcon } from '@alfalab/icons-classic';

import { CameraMIcon } from '@alfalab/icons-glyph';

import { VisaXxlIcon, MastercardLIcon, MirXxlIcon } from '@alfalab/icons-logotype';

import styles from './index.module.css';
import { validateCardNumber } from './utils';

export type BankCardProps = {
    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Цвет фона карты
     */
    backgroundColor?: string;

    /**
     * Иконка логотипа банка (размер L)
     */
    bankLogo?: ReactNode;

    /**
     * Лэйбл поля ввода
     */
    inputLabel?: string;

    /**
     * Значение поля ввода
     */
    value?: string;

    /**
     * Обработчик ввода
     */
    onChange?: (event: ChangeEvent<HTMLInputElement>, payload: { value: string }) => void;

    /**
     * Обработчик вызова камеры
     */
    onUsePhoto?: (event: MouseEvent<HTMLButtonElement>) => void;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
};

// prettier-ignore
const cardMask = [/\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/];
// prettier-ignore
const accountNumberMask = [/\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/, ' ', /\d/, /\d/, /\d/, /\d/];

const getBrandIcon = (value = '') => {
    // Показываем логотип только после ввода всех цифр карты
    if (value.replace(/\s/g, '').length === 16 && validateCardNumber(value)) {
        if (value.startsWith('2')) return <MirXxlIcon />;
        if (value.startsWith('4')) return <VisaXxlIcon />;
        if (value.startsWith('5')) return <MastercardLIcon />;
        if (value.startsWith('6')) return <MastercardLIcon />;
    }
    return null;
};

export const BankCard = React.forwardRef<HTMLInputElement, BankCardProps>(
    (
        {
            bankLogo = <BankAlfaLColorIcon />,
            backgroundColor = '#EF3124',
            inputLabel = 'Номер карты или счёта',
            value,
            className,
            onUsePhoto,
            onChange,
            dataTestId,
        },
        ref,
    ) => {
        const uncontrolled = value === undefined;

        const [brandIcon, setBrandIcon] = useState<ReactNode>(getBrandIcon(value));

        const getMask = (newValue: string) =>
            newValue.length <= cardMask.length ? cardMask : accountNumberMask;

        const handleInputChange = useCallback(
            (event: ChangeEvent<HTMLInputElement>, payload) => {
                if (uncontrolled) {
                    setBrandIcon(getBrandIcon(event.target.value));
                }

                if (onChange) {
                    onChange(event, payload);
                }
            },
            [onChange, uncontrolled],
        );

        const renderRightAddons = useCallback(
            () => (
                <button type='button' className={styles.usePhoto} onClick={onUsePhoto}>
                    <CameraMIcon />
                </button>
            ),
            [onUsePhoto],
        );

        useEffect(() => {
            setBrandIcon(getBrandIcon(value));
        }, [value]);

        return (
            <div className={cn(styles.component, className)}>
                <div className={styles.aspectRatioContainer}>
                    <div className={styles.content} style={{ backgroundColor }}>
                        <div className={styles.bankLogo}>{bankLogo}</div>

                        <MaskedInput
                            ref={ref}
                            value={value}
                            mask={getMask}
                            block={true}
                            label={inputLabel}
                            rightAddons={renderRightAddons()}
                            inputClassName={styles.input}
                            labelClassName={styles.label}
                            filledClassName={styles.filled}
                            focusedClassName={styles.focused}
                            onChange={handleInputChange}
                            dataTestId={dataTestId}
                            inputMode='numeric'
                            pattern='[0-9]*'
                        />

                        {brandIcon && <div className={styles.brandLogo}>{brandIcon}</div>}
                    </div>
                </div>
            </div>
        );
    },
);

BankCard.defaultProps = {
    bankLogo: <BankAlfaLColorIcon />,
    backgroundColor: '#EF3124',
    inputLabel: 'Номер карты или счёта',
};
