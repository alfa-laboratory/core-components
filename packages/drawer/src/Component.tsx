import React, { forwardRef, ReactNode, useMemo } from 'react';
import cn from 'classnames';

import { BaseModal, BaseModalProps } from '@alfalab/core-components-modal';

import styles from './index.module.css';

export const ANIMATION_DURATION = 600;

export type DrawerProps = Omit<
    BaseModalProps,
    | 'onHeaderHighlight'
    | 'onFooterHighlight'
    | 'highlightHeader'
    | 'highlightFooter'
    | 'header'
    | 'hideBackdrop'
    | 'backdropProps'
    | 'fullscreen'
    | 'container'
    | 'targetHandleExited'
    | 'Transition'
    | 'transitionProps'
> & {
    /**
     * Шапка
     */
    header?: ReactNode;

    /**
     * Футер
     */
    footer?: ReactNode;
};

export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(
    ({ open, className, children, header, footer, contentClassName, ...restProps }, ref) => {
        const transitionProps = useMemo(
            () => ({
                classNames: styles,
                timeout: ANIMATION_DURATION,
            }),
            [],
        );

        const backdropProps = useMemo(
            () => ({
                classNames: {
                    enter: styles.backdropEnter,
                    enterActive: styles.backdropEnterActive,
                    appear: styles.backdropAppear,
                    appearActive: styles.backdropAppearActive,
                    exit: styles.backdropExit,
                    exitActive: styles.backdropExitActive,
                },
                timeout: ANIMATION_DURATION,
            }),
            [],
        );

        return (
            <BaseModal
                {...restProps}
                ref={ref}
                open={open}
                className={cn(styles.component, className)}
                contentClassName={cn(contentClassName, styles.content)}
                fullscreen={true}
                targetHandleExited='children'
                transitionProps={transitionProps}
                backdropProps={backdropProps}
                header={header}
                footer={footer}
            >
                {children}
            </BaseModal>
        );
    },
);
