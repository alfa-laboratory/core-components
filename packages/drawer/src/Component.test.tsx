import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Drawer } from './Component';

describe('Drawer', () => {
    let savedBodyStyle: CSSStyleDeclaration;

    beforeAll(() => {
        savedBodyStyle = document.body.style;
    });

    beforeEach(() => {
        // eslint-disable-next-line
        // @ts-ignore
        document.body.setAttribute('style', savedBodyStyle);
    });

    afterAll(() => {
        cleanup();
    });

    it('should match snapshot', () => {
        const { queryByTestId } = render(
            <Drawer
                header={<span>header</span>}
                footer={<span>footer</span>}
                open={true}
                dataTestId='drawer'
            >
                content
            </Drawer>,
        );

        expect(queryByTestId('drawer')).toMatchSnapshot();
    });

    it('should not render anything when open=false', () => {
        const { queryByTestId } = render(
            <Drawer open={false} dataTestId='drawer'>
                content
            </Drawer>,
        );

        expect(queryByTestId('drawer')).not.toBeInTheDocument();
    });
});
