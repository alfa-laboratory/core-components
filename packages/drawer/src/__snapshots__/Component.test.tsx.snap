// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should match snapshot 1`] = `
<div
  class="wrapper"
  data-test-id="drawer"
  role="dialog"
  style="z-index: 100;"
  tabindex="-1"
>
  <div
    class="component enter enterActive"
  >
    <div
      class="content"
    >
      <div
        class="content contentEnter contentEnterActive"
      >
        <span>
          header
        </span>
        <span>
          content
        </span>
        <span>
          footer
        </span>
      </div>
    </div>
  </div>
</div>
`;
